var/spack/repos/builtin/packages/mercury/package.py

```diff
@@ -38,78 +38,63 @@ class Mercury(CMakePackage):
-            '-DBUILD_SHARED_LIBS=ON',
+
+        def variant_bool(feature, on='ON', off='OFF'):
+            """Ternary for spec variant to ON/OFF string"""
+            if feature in spec:
+                return on
+            return off
+
+        cmake_args = [
+            '-DBUILD_SHARED_LIBS=%s' % variant_bool('+shared'),
+            '-DMERCURY_USE_BOOST_PP:BOOL=ON',
             '-DMERCURY_USE_CHECKSUMS=ON',
             '-DMERCURY_USE_EAGER_BULK=ON',
+            '-DMERCURY_USE_SELF_FORWARD=%s' % variant_bool('+selfforward'),
             '-DMERCURY_USE_SYSTEM_MCHECKSUM=OFF',
             '-DMERCURY_USE_XDR=OFF'
+            '-DNA_USE_BMI:BOOL=%s' % variant_bool('+bmi'),
+            '-DNA_USE_CCI:BOOL=%s' % variant_bool('+cci'),
+            '-DNA_USE_MPI:BOOL=%s' % variant_bool('+mpi'),
+            '-DNA_USE_SM:BOOL=%s'  % variant_bool('+sm'),
         ]
 
-        if '+boost' in spec:
-            args.append('-DMERCURY_USE_BOOST_PP=ON')
-        else:
-            args.append('-DMERCURY_USE_BOOST_PP=OFF')
-
-        if '+bmi' in spec:
-            args.append('-DNA_USE_BMI=ON')
-        else:
-            args.append('-DNA_USE_BMI=OFF')
-
-        if '+cci' in spec:
-            args.append('-DNA_USE_CCI=ON')
-        else:
-            args.append('-DNA_USE_CCI=OFF')
-
-        if '+mpi' in spec:
-            args.append('-DNA_USE_MPI=ON')
-        else:
-            args.append('-DNA_USE_MPI=OFF')
+        if '@1.0.0:' in spec:
+            cmake_args.extend([
+                '-DMERCURY_USE_SYSTEM_BOOST:BOOL=%s'
+                % variant_bool('+boostsys'),
+                '-DNA_USE_OFI:BOOL=%s' % variant_bool('+ofi'),
+            ])
 
         if '+ofi' in spec:
-            args.append('-DNA_USE_OFI=ON')
+            cmake_args.extend([
+                '-DNA_OFI_GNI_USE_UDREG=%s' % variant_bool('+udreg'),
+            ])
             if self.run_tests:
-                args.append('-DNA_OFI_TESTING_PROTOCOL={0}'.format(';'.join(
-                    spec['libfabric'].variants['fabrics'].value)))
-        else:
-            args.append('-DNA_USE_OFI=OFF')
-
-        if '+sm' in spec:
-            args.append('-DNA_USE_SM=ON')
-        else:
-            args.append('-DNA_USE_SM=OFF')
-
-        if '+opa' in spec:
-            args.append('-DMERCURY_USE_OPA=ON')
-        else:
-            args.append('-DMERCURY_USE_OPA=OFF')
-
-        if '+selfforward' in spec:
-            args.append('-DMERCURY_USE_SELF_FORWARD=ON')
-        else:
-            args.append('-DMERCURY_USE_SELF_FORWARD=OFF')
-
-        if '+udreg' in spec:
-            args.append('-DNA_OFI_GNI_USE_UDREG=ON')
-        else:
-            args.append('-DNA_OFI_GNI_USE_UDREG=OFF')
+                cmake_args.extend('-DNA_OFI_TESTING_PROTOCOL={0}'.format(
+                    ';'.join(spec['libfabric'].variants['fabrics'].value)))
 
         if self.run_tests:
-            args.append('-DBUILD_TESTING=ON')
+            args.extend('-DBUILD_TESTING=ON')
```

Comment:
Shouldn't these be `cmake_args.append('-DFOO=BAR')`?  If I'm reading the diff correctly the variable name changed from `args` to `cmake_args` and `append` takes a single value while `extend` expects a list.

Reply:
Also it seems like these could use the `variant_bool` helper function you defined above.

Reply:
ok thanks I fixed that one.